docs/switch.rst

```diff
@@ -0,0 +1,369 @@
+=====================================
+Switching from GDAL's Python bindings
+=====================================
+
+Switching from GDAL's Python bindings, meaning the ``gdal`` module in the
+``osgeo`` namespace, to Rasterio for new projects or new iterations on existing
+projects may not be complicated. This document explains the key similarities
+and differences between these two Python packages and highlights the features
+of Rasterio that can help in switching.
```

Comment:
The first sentence has an difficult cadence and could be trimmed down: specifically we could just call it the "`osgeo.gdal` module" and I don't see a need to differentiate new vs. existing projects. 

> Switching from GDAL's Python bindings (the `osgeo.gdal` module) to Rasterio may not be complicated.

Additionally, I think we need a sentence in here to answer, "Why would someone  care to undertake the effort? What does a user have to gain by switching?"